sources/api/schnauzer/src/helpers.rs

```diff
@@ -592,6 +612,123 @@ pub fn join_array(
     Ok(())
 }
 
+/// calculate_memory_mebibytes_to_reserve and calculate_cpu_millicores_to_reserve are taken from EKS' calculations.
+/// https://github.com/awslabs/amazon-eks-ami/blob/db28da15d2b696bc08ac3aacc9675694f4a69933/files/bootstrap.sh
+
+/// Calculates the amount of memory to reserve for kubeReserved in mebibytes.
+/// Formula: memory_to_reserve = max_num_pods * 11 + 255 is taken from
+/// https://github.com/awslabs/amazon-eks-ami/pull/419#issuecomment-609985305
+pub fn calculate_memory_mebibytes_to_reserve(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    trace!("Starting calculate_memory_mebibytes_to_reserve helper");
+    let template_name = template_name(renderctx);
+    trace!("Template name: {}", &template_name);
+
+    trace!("Number of params: {}", helper.params().len());
+    check_param_count(helper, template_name, 2)?;
+
+    let max_num_pods_val = get_param(helper, 0)?;
+    let max_num_pods = match max_num_pods_val {
+        Value::Bool(b) => b.to_string(),
+        Value::Number(n) => n.to_string(),
+        Value::String(s) => s.to_string(),
+
+        Value::Null | Value::Array(_) | Value::Object(_) => {
+            return Err(RenderError::from(
+                error::TemplateHelperError::InvalidTemplateValue {
+                    expected: "scalar",
+                    value: max_num_pods_val.to_owned(),
+                    template: template_name.to_owned(),
+                },
+            ))
+        }
+    };
```

Comment:
Can we reject everything but the `Value::Number(n)` case, and convert it directly to a `u32`?

We do that elsewhere with keys that we know should be strings.

Reply:
Yeah we can reject everthing but `Value::Number(n)`; however, we can not convert it to `u32`. Because the type of this value is `&serde_json::Number`, so I can not phase it to `u32` directly. My solution is about converting it to `string` here, and then phase it to `u32`. If my solution is not that efficient, please correct me.